jax/lax/lax_parallel.py

```diff
@@ -497,12 +496,28 @@ def _ppermute_batcher(vals_in, dims_in, axis_size, axis_name, perm):
 batching.collective_rules[ppermute_p] = _ppermute_batcher
 
 
+def _all_to_all_via_psum(x, *, axis_name, split_axis, concat_axis):
+  zero_full = lax.full((x.shape[split_axis],) + x.shape, 0, dtype=x.dtype)
+  idx = axis_index(axis_name)
+  local_full = lax.dynamic_update_slice_in_dim(zero_full, x[None], idx, 0)
+  global_full = psum(local_full, axis_name=axis_name)
```

Comment:
I'm not sure, but can you call `_allgather` here? It seems similar.

Reply:
Nice catch, I used `all_gather` in the latest implementation